lib/block-supports/custom-css.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Custom CSS block support.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Stores the custom CSS for block instances to be output in the footer.
+ *
+ * @var string
+ */
+global $gutenberg_block_custom_css;
+$gutenberg_block_custom_css = '';
+
+/**
+ * Get the custom CSS class name for a block.
+ *
+ * @since 7.0.0
+ *
+ * @param array $block Block object.
+ * @return string The custom CSS class name.
+ */
+function gutenberg_get_custom_css_class_name( $block ) {
+	return wp_unique_id_from_values( $block, 'wp-custom-css-' );
+}
+
+/**
+ * Render the custom CSS stylesheet and add class name to block as required.
+ *
+ * @since 7.0.0
+ *
+ * @param array $parsed_block The parsed block.
+ * @return array The same parsed block with custom CSS class name added if appropriate.
+ */
+function gutenberg_render_custom_css_support_styles( $parsed_block ) {
+	global $gutenberg_block_custom_css;
+
+	$custom_css = $parsed_block['attrs']['style']['css'] ?? null;
+
+	if ( empty( $custom_css ) ) {
+		return $parsed_block;
+	}
+
+	// Generate a unique class name for this block instance.
+	$class_name         = gutenberg_get_custom_css_class_name( $parsed_block );
```

Comment:
```suggestion
	$class_name         = wp_unique_id_from_values( $parsed_block 'wp-custom-css-' );
```

We could call the method directly here; there's no need to introduce a new function, mainly when it's used only once.